src/Microsoft.OData.Client/SaveResult.cs

```diff
@@ -476,12 +476,12 @@ private ODataRequestMessageWrapper CreateNextRequest()
                     EntityDescriptor entityDescriptor = (EntityDescriptor)descriptor;
 
                     if (((EntityStates.Unchanged == descriptor.State) || (EntityStates.Modified == descriptor.State)) &&
-                        (null != (requestMessage = this.CheckAndProcessMediaEntryPut(entityDescriptor))))
+                        ((requestMessage = this.CheckAndProcessMediaEntryPut(entityDescriptor)) != null))
                     {
                         this.streamRequestKind = StreamRequestKind.PutMediaResource;
                         descriptorForSendingRequest2 = entityDescriptor.DefaultStreamDescriptor;  // We want to give SendingRequest2 the StreamDescriptor
                     }
-                    else if ((EntityStates.Added == descriptor.State) && (null != (requestMessage = this.CheckAndProcessMediaEntryPost(entityDescriptor))))
+                    else if ((EntityStates.Added == descriptor.State) && ((requestMessage = this.CheckAndProcessMediaEntryPost(entityDescriptor)) != null))

```

Comment:
@xuzhg I think this `(EntityStates.Added == descriptor.State)` should change to `(descriptor.State == EntityStates.Added)`

Reply:
Changed.